### ISSUE TITLE
House Rule: "Essence loss only reduces MAG/RES Maximum" broken

### ISSUE THREAD

**Issue by Emparawr** [chummer5a/chummer5a#3440]
So the House Rule, "Essence loss only reduces MAG/RES Maximum" works properly at first, but when I save the character, close it, close chummer, and reopen both, it has reduced the maximum and current Magic Attribute instead of just the maximum. The only way to fix it is to remove the cyberware, then reapply it. The problem is that once you save and close and reload the character, it's broken again. 

### Environment
Chummer Version: 5.204.0
Environment: Windows 10

### Expected Behavior
That with the House Rule: "Essence loss only reduces MAG/RES Maximum", option checked and the option properly saved, Essence loss should only reduce the maximum magic attribute, not the current magic attribute, assuming the initial current magic attribute is lower than the initial maximum magic attribute.  

### Actual behaviour
When the character sheet is reloaded after having been saved, both the current and maximum magic attribute are reduced by 1. 

### Steps to reproduce the behavior
1) Open Chummer.
2) Go to the tools menu, then Options, then the House Rules tab, and check the "Essence loss only reduces MAG/RES Maximum" option. 
3) Close Chummer
4) Reopen Chummer
5) Create a new character with the magic priority set to C, and choose the "Adept - 4 Magic" option
6) Apply between .1 and 1 essence worth of cyberware or bioware. 
7) The current magic attribute stays at 4, and the maximum magic attribute is appropriately reduced to 5 from 6. 
8) Save the Character file. 
9) Close Chummer
10) Open Chummer
11) Open Character File
12) Notice that both the current and maximum magic attributes are being reduced by 1, instead of only reducing the maximum.

**Comment by DelnarErsike** [comment]
I cannot seem to be able to replicate this issue on the latest dev build. Could you test to see if this issue is still present in the latest nightly, and if so, provide a character's savefile that we could load to replicate it?

**Comment by Mliekseter** [comment]
I got this issue too so I did some playing around. It seems to occur for me only when both the "Essence loss only reduces MAG/RES Maximum" and "Karma cost for increasing Special Attributes is not reduced with Essence Loss" houserules are ticked on.
Also, from looking at the save files themselves, it seems the buggy save and the working save have different improvements for Essence loss. In the working save, the "EssenceLossChargen" improvement applies <max>-1<max>, whereas in the buggy save it applies <aug>-1<aug> instead.
